### PR TITLE
Bugfix-44451 : Throw error message for the given condition

### PR DESCRIPTION
**Pull request by say1bt**
User cannot save if the field name is same as the current question name in expression builder.

### CODE REVIEW

File: ValidationRules/src/Features/index.tsx [hunk 1]
```diff
@@ -48,18 +59,45 @@ const ParentComponent = ({
   const [sections, setSections] = useState<any[]>([]);
   const [defaultSections, setDefaultSections] = useState<any[]>([]);
   const [_nestedRows, _setNestedRows] = useState<any>([]);
+  // const [_nestedRows, _setNestedRows] = useState<any>([
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:62]: remove those commented lines

File: ValidationRules/src/Features/index.tsx [hunk 2]
```diff
@@ -71,48 +109,59 @@ const ParentComponent = ({
...
-  //   const [currentQuestionDetails, setCurrentQuestionDetails] = useState<any>({
-  //     "label": "TSDTem_C01_S01_date",
-  //     "value": "TSDTem_C01_S01_date",
-  //     "questionType": "List",
-  //     "questionId": "b76bc889-6d66-ee11-9ae7-6045bdd0ef22",
-  //     "questionLabel": "date"
+  const [currentListQuestionAnswers, setCurrentListQuestionAnswers] =
+    useState();
+
+  // const [currentQuestionDetails, setCurrentQuestionDetails] = useState<any>({
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:125]: remove those comments

File: ValidationRules/src/Features/index.tsx [hunk 3]
```diff
@@ -180,18 +240,30 @@ const ParentComponent = ({
...
-      currentPossitionDetails?.currentPosition === 'question' &&
+      currentPossitionDetails?.currentPosition === "question" &&
       !localTest
     ) {
       const currnetQuestionDetails = questionList?.find(
-        (ques) => ques?.questionId === currentPossitionDetails?.id?.toLowerCase()
+        (ques) =>
+          ques?.questionId === currentPossitionDetails?.id?.toLowerCase()
+      );
+      console.log(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:250]: remove consoles

File: ValidationRules/src/Features/index.tsx [hunk 4]
```diff
@@ -180,18 +240,30 @@ const ParentComponent = ({
...
+        "currentPossitionDetails in useEffect",
+        currentPossitionDetails
+      );
+
+      console.log(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:255]: remove consoles

File: ValidationRules/src/Features/index.tsx [hunk 5]
```diff
@@ -305,15 +381,15 @@ const ParentComponent = ({
                         checkBoxValues: [
                           {
                             show: {
-                              logicalName: 'Show',
-                              value: 'show'
-                            }
-                          }
-                        ]
-                      }
+                              logicalName: "Show",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:384]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 6]
```diff
@@ -388,19 +468,22 @@ const ParentComponent = ({
...
                         checkBoxValues: [
                           {
-                            'OutPutDoc:Show': {
-                              logicalName: 'Show in Document',
-                              value: 'OutPutDoc:Show'
-                            }
-                          }
-                        ]
-                      }
+                            "OutPutDoc:Show": {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:471]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 7]
```diff
@@ -475,18 +571,18 @@ const ParentComponent = ({
                     actions: [
                       {
                         minMax: {
-                          logicalName: 'minMax',
+                          logicalName: "minMax",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:574]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 8]
```diff
@@ -520,25 +616,25 @@ const ParentComponent = ({
               let actionMap: any;
               if (!defaultAction && defaultAction !== 0) {
                 actionMap = {
-                  type: 'CLE_Q',
-                  value: null
+                  type: "CLE_Q",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:619]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 9]
```diff
@@ -520,25 +616,25 @@ const ParentComponent = ({
...
+                  value: null,
                 };
-              } else if (typeof defaultAction === 'object') {
+              } else if (typeof defaultAction === "object") {
                 if (defaultAction?.var) {
                   actionMap = {
-                    type: 'VAL_Q',
-                    value: defaultAction?.var
+                    type: "VAL_Q",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:625]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 10]
```diff
@@ -520,25 +616,25 @@ const ParentComponent = ({
...
+                    value: defaultAction?.var,
                   };
                 } else {
                   actionMap = {
-                    type: 'MAT_F',
-                    value: defaultAction
+                    type: "MAT_F",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:630]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 11]
```diff
@@ -520,25 +616,25 @@ const ParentComponent = ({
...
+                    value: defaultAction,
                   };
                 }
               } else {
                 actionMap = {
-                  type: 'ADD_V',
-                  value: defaultAction
+                  type: "ADD_V",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:636]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 12]
```diff
@@ -602,7 +702,7 @@ const ParentComponent = ({
 
   // This useEffect is responsible for Convert DB Format to our JSON format
   useEffect(() => {
-    console.log('_visibilityAndDocOutput', _visibilityAndDocOutput);
+    console.log("_visibilityAndDocOutput", _visibilityAndDocOutput);
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:705]: if this console is not useful then remove it

File: ValidationRules/src/Features/index.tsx [hunk 13]
```diff
@@ -613,7 +713,8 @@ const ParentComponent = ({
         )
           return;
         _setNestedRows((prevData: any) => {
-          let visibilityAndDocRuleOutput = dbData?.visibilityAndDocRuleOutput?.if?.length
+          let visibilityAndDocRuleOutput = dbData?.visibilityAndDocRuleOutput
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:716]: if this value not change after define put it as const

File: ValidationRules/src/Features/index.tsx [hunk 14]
```diff
@@ -665,25 +778,28 @@ const ParentComponent = ({
                       {
                         checkBoxValues: [
                           {
-                            'OutPutDoc:Show': {
-                              logicalName: 'Show in Document',
-                              value: 'OutPutDoc:Show'
-                            }
+                            "OutPutDoc:Show": {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:781]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 15]
```diff
@@ -708,19 +824,19 @@ const ParentComponent = ({
     let defaultValueRule: any;
     setIsApiDataLoaded(false);
     let logicalName;
-    if (currentPossitionDetails?.currentPosition === 'chapter') {
+    if (currentPossitionDetails?.currentPosition === "chapter") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:827]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 16]
```diff
@@ -708,19 +824,19 @@ const ParentComponent = ({
...
       logicalName = dbConstants.chapter.fieldName;
-    } else if (currentPossitionDetails?.currentPosition === 'section') {
+    } else if (currentPossitionDetails?.currentPosition === "section") {
       logicalName = dbConstants.section.fieldName;
-    } else if (currentPossitionDetails?.currentPosition === 'question') {
+    } else if (currentPossitionDetails?.currentPosition === "question") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:831]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 17]
```diff
@@ -708,19 +824,19 @@ const ParentComponent = ({
...
       logicalName = dbConstants.question.fieldName;
     }
 
     if (
       logicalName &&
       currentPossitionDetails?.id &&
-      (currentPossitionDetails?.currentPosition === 'chapter' ||
-        currentPossitionDetails?.currentPosition === 'section')
+      (currentPossitionDetails?.currentPosition === "chapter" ||
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:838]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 18]
```diff
@@ -730,7 +846,7 @@ const ParentComponent = ({
     } else if (
       logicalName &&
       currentPossitionDetails?.id &&
-      currentPossitionDetails?.currentPosition === 'question'
+      currentPossitionDetails?.currentPosition === "question"
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:849]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 19]
```diff
@@ -874,131 +1009,153 @@ const ParentComponent = ({
   ) => {
     let logicalName;
 
-    if (currentPossitionDetails?.currentPosition === 'question') {
+    if (currentPossitionDetails?.currentPosition === "question") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:1012]: add hard coded values as constants

File: ValidationRules/src/Features/index.tsx [hunk 20]
```diff
@@ -874,131 +1009,153 @@ const ParentComponent = ({
...
       logicalName = dbConstants.question.fieldName;
-    } else if (currentPossitionDetails?.currentPosition === 'section') {
+    } else if (currentPossitionDetails?.currentPosition === "section") {
       logicalName = dbConstants.section.fieldName;
-    } else if (currentPossitionDetails?.currentPosition === 'chapter') {
+    } else if (currentPossitionDetails?.currentPosition === "chapter") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:1016]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 21]
```diff
@@ -161,14 +184,14 @@ const RowContainer: React.FC<TableRowProps> = ({
         higestLevel = idGenerator(level, hasNested, []);
       }
       let newRow = {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:186]: define this in common file as common variable

File: ValidationRules/src/Features/rowContainer.tsx [hunk 22]
```diff
@@ -197,16 +222,17 @@ const RowContainer: React.FC<TableRowProps> = ({
               {
                 fieldName: fieldValue?.fieldName,
                 fieldValue:
-                  typeof fieldValue?.input === 'string'
+                  typeof fieldValue?.input === "string"
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:225]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 23]
```diff
@@ -225,44 +251,64 @@ const RowContainer: React.FC<TableRowProps> = ({
       fieldValueSetToNestedRows(fieldValue);
     }
     // Added validation, If the same level expression Changed other child has to be changed
-    if (fieldValue?.fieldName === 'expression') {
-      let releatedFields = _nestedRows?.find((x: any[]) => x[sectionLevel])?.[sectionLevel]?.fields;
+    if (fieldValue?.fieldName === "expression") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:254]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 24]
```diff
@@ -225,44 +251,64 @@ const RowContainer: React.FC<TableRowProps> = ({
...
             const firstExp = releatedFields[1]?.expression;
             const initialEmptyFieldId = releatedFields[0]?.level;
-            parentIds = parentIds?.filter((item: any) => item !== initialEmptyFieldId);
+            parentIds = parentIds?.filter(
+              (item: any) => item !== initialEmptyFieldId
+            );
             if (firstExp !== fieldValue.input && parentIds.length) {
-              openNotificationWithIcon('error', 'First Selected Expression cannot be changed!');
+              openNotificationWithIcon(
+                "error",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:300]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 25]
```diff
@@ -271,59 +317,80 @@ const RowContainer: React.FC<TableRowProps> = ({
...
       let higestLevel;
       if (nearestIdParentObject?.innerConditions?.length)
         higestLevel = Math.max(
-          ...nearestIdParentObject.innerConditions.map((x: { level: any }) => x.level)
+          ...nearestIdParentObject.innerConditions.map(
+            (x: { level: any }) => x.level
+          )
         );
 
       let newRow = {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:376]: get this from previously i mentioned file

File: ValidationRules/src/Features/rowContainer.tsx [hunk 26]
```diff
@@ -375,30 +458,38 @@ const RowContainer: React.FC<TableRowProps> = ({
     _setNestedRows(
       updateAllLevelArray(_nestedRows, sectionLevel, [
         {
-          field: '',
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:378]: reffer previous mentions value

File: ValidationRules/src/Features/rowContainer.tsx [hunk 27]
```diff
@@ -469,7 +571,7 @@ const RowContainer: React.FC<TableRowProps> = ({
         setIsLoad(false);
       }
     } catch (error) {
-      console.error('Error fetching data:', error);
+      console.error("Error fetching data:", error);
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:574]: handle this error properly

File: ValidationRules/src/Features/rowContainer.tsx [hunk 28]
```diff
@@ -573,53 +702,55 @@ const RowContainer: React.FC<TableRowProps> = ({
...
-            var: minMax?.maxValue
+            var: minMax?.maxValue,
           };
         }
         const formattingForMin = [];
         const formattingForMax = [];
         formattingForMin.push(
-          _minMaxDbFormarFields?.exp['']?.length
-            ? _minMaxDbFormarFields?.exp[''][0]
+          _minMaxDbFormarFields?.exp[""]?.length
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:718]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 29]
```diff
@@ -573,53 +702,55 @@ const RowContainer: React.FC<TableRowProps> = ({
...
+          currentQuestionDetails?.questionType ===
+            dbConstants.questionTypes.stringQuestion ||
+          currentQuestionDetails?.questionType ===
+            dbConstants.questionTypes.dateTimeQuestion
         ) {
           minMaxDBFormatArray.push(
             {
-              type: 'MINIMUM_LENGTH',
-              value: { if: formattingForMin }
+              type: "MINIMUM_LENGTH",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:737]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 30]
```diff
@@ -573,53 +702,55 @@ const RowContainer: React.FC<TableRowProps> = ({
...
+              value: { if: formattingForMin },
             },
             {
-              type: 'MAXIMUM_LENGTH',
-              value: { if: formattingForMax }
+              type: "MAXIMUM_LENGTH",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:741]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 31]
```diff
@@ -573,53 +702,55 @@ const RowContainer: React.FC<TableRowProps> = ({
...
+              value: { if: formattingForMax },
             }
           );
         } else {
           minMaxDBFormatArray.push(
             {
-              type: 'MINIMUM',
-              value: { if: formattingForMin }
+              type: "MINIMUM",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:748]: add hard coded values as constants

File: ValidationRules/src/Features/rowContainer.tsx [hunk 32]
```diff
@@ -573,53 +702,55 @@ const RowContainer: React.FC<TableRowProps> = ({
...
+              value: { if: formattingForMin },
             },
             {
-              type: 'MAXIMUM',
-              value: { if: formattingForMax }
+              type: "MAXIMUM",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/rowContainer.tsx:752]: add hard coded values as constants

**Review by tilangapramithvidusara — CHANGES_REQUESTED** [review]

**Review by anujshashimal1bt — APPROVED** [review]

**Pull request merged**